### PR TITLE
Use TIFFGetFieldDefaulted and remove warning about falling back to GenericImage decoder

### PR DESCRIPTION
**Pull request by jantonguirao**
Signed-off-by: Joaquin Anton <janton@nvidia.com>

#### Why we need this PR?
*Pick one, remove the rest*
- It fixes a bug when decoding TIFF images without TIFFTAG_SAMPLESPERPIXEL, TIFFTAG_BITSPERSAMPLE in the header.

#### What happened in this PR?
*Fill relevant points, put NA otherwise. Replace anything inside []*
 - What solution was applied:
     *Use TIFFGetFieldDefaulted to use default values in non-mandatory tags*
     *Remove warning about falling back to GenericImage decoder*
 - Affected modules and functionalities:
     *ImageDecoder*
 - Key points relevant for the review:
     *All of it*
 - Validation and testing:
     *Existing tests*
 - Documentation (including examples):
     *N/A*

Depends on https://github.com/NVIDIA/DALI_extra/pull/41

**JIRA TASK**: *[DALI-1541]*

### CODE REVIEW

File: dali/image/tiff_libtiff.cc [hunk 1]
```diff
@@ -247,17 +247,19 @@ TiffImage_Libtiff::TiffImage_Libtiff(const uint8_t *encoded_buffer,
   LIBTIFF_CALL(
     TIFFGetField(tif_.get(), TIFFTAG_IMAGEWIDTH, &shape_[1]));
   LIBTIFF_CALL(
-    TIFFGetField(tif_.get(), TIFFTAG_SAMPLESPERPIXEL, &shape_[2]));
+    TIFFGetFieldDefaulted(tif_.get(), TIFFTAG_SAMPLESPERPIXEL, &shape_[2]));
```
Comment (JanuszL) [inline dali/image/tiff_libtiff.cc:250]: Can you add a test for it? Maybe add an additional TIFF image with that properites to DALI_extra?

Comment (jantonguirao) [reply dali/image/tiff_libtiff.cc:250]: Good idea

**Comment by jantonguirao** [comment]
!build

**Comment by jantonguirao** [comment]
!build

**Comment by jantonguirao** [comment]
!build

**Comment by jantonguirao** [comment]
!build

**Comment by jantonguirao** [comment]
!build

File: qa/setup_dali_extra.sh [hunk 2]
```diff
@@ -2,11 +2,12 @@
...
 # Fetch test data
 export DALI_EXTRA_PATH=${DALI_EXTRA_PATH:-/opt/dali_extra}
-export DALI_EXTRA_URL=${DALI_EXTRA_URL:-"https://github.com/NVIDIA/DALI_extra.git"}
+export DALI_EXTRA_URL="https://github.com/jantonguirao/DALI_extra.git"
 
 DIR=$(cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null && pwd )
-DALI_EXTRA_VERSION_PATH="${DIR}/../DALI_EXTRA_VERSION"
-read -r DALI_EXTRA_VERSION < ${DALI_EXTRA_VERSION_PATH}
+#DALI_EXTRA_VERSION_PATH="${DIR}/../DALI_EXTRA_VERSION"
+#read -r DALI_EXTRA_VERSION < ${DALI_EXTRA_VERSION_PATH}
```
Comment (mzient) [inline qa/setup_dali_extra.sh:9]: Why?

Comment (jantonguirao) [reply qa/setup_dali_extra.sh:9]: Testing with a new image that I added to DALI_extra, I'm going to remove it before merging

File: qa/setup_dali_extra.sh [hunk 3]
```diff
@@ -2,11 +2,12 @@
 
 # Fetch test data
```
Comment (jantonguirao) [inline qa/setup_dali_extra.sh:3]: The changes in this file are going to be reverted

**Comment by jantonguirao** [comment]
!build

**Pull request merged**